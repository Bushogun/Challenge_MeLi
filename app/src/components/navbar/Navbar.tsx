import React from 'react';
import Image from 'next/image';
import Link from 'next/link';
import { SearchBarForm } from '@/src/components/navbar/search-bar-form/search-bar-form';
import styles from './navbar.module.scss';
import { useRouter } from 'next/router';

export const Navbar = () => {
  const router = useRouter();

  const handleLogoClick = () => {
    router.reload();
  };

  return (
    <nav className={styles.navbar}>
      <Link href="/">
        <Image
          src="/logo_ml.png"
          alt="Logo Mercado Libre"
          title="Logo Mercado Libre"
          width="60"
          height="45"
          priority={true}
          onClick={handleLogoClick}
        />
      </Link>
      <SearchBarForm />
    </nav>
  );
};
