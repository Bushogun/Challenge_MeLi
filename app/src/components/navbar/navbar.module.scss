@use '../../styles/colors.module.scss' as *;

.navbar{
	background-color: $color__primary;
	display: flex;
	align-items: center;
	justify-content: center;
    padding: 5px;
	a {
		padding-right: 2rem;
	  }
}