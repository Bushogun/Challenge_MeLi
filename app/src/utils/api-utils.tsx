import { setLoading } from '@/src/store/productSlice';

export async function fetchProductData(
  dispatch: any,
  searchQuery: string,
  limit: string,
  selectedSort: string,
  priceFilter: string,
) {
  try {
    dispatch(setLoading(true));

    let sortQueryParam = '';
    let priceQueryParam = '';
    if (selectedSort) {
      sortQueryParam = `&sort=${selectedSort}`;
    }
    if (priceFilter) {
      priceQueryParam = `&price=${priceFilter}`;
    }

    const response = await fetch(
      `https://api.mercadolibre.com/sites/MLA/search?q=${searchQuery}&limit=${limit}${sortQueryParam}${priceQueryParam}`
    );
    console.log (response);
    if (!response.ok) {
      throw new Error('Error al cargar productos');
    }

    const data = await response.json();
    dispatch(setLoading(false));

    return data;
  } catch (error) {
    dispatch(setLoading(false));
    throw error;
  }
}
