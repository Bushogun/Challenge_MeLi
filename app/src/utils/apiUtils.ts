
export async function fetchProductData(
    searchQuery: string,
    limit: number,
    selectedSort: string | null,
    priceFilter: string | null
  ) {
    const sortQueryParam = selectedSort ? `&sort=${selectedSort}` : '';
    const priceQueryParam = priceFilter ? `&price=${priceFilter}` : '';
  
    const response = await fetch(
      `https://api.mercadolibre.com/sites/MLA/search?q=${searchQuery}&limit=${limit}${sortQueryParam}${priceQueryParam}`
    );
  
    if (!response.ok) {
      throw new Error('Error al cargar productos');
    }
  
    const data = await response.json();
    return data;
  }

/*
export async function fetchProducts(
    searchQuery: string,
    limit: number,
    selectedSort: string | null,
    priceFilter: string | null
  ) {
    const sortQueryParam = selectedSort ? `&sort=${selectedSort}` : '';
    const priceQueryParam = priceFilter ? `&price=${priceFilter}` : '';
  
    const response = await fetch(
      `https://api.mercadolibre.com/sites/MLA/search?q=${searchQuery}&limit=${limit}${sortQueryParam}${priceQueryParam}`
    );
  
    if (!response.ok) {
      throw new Error('Error al cargar productos');
    }
  
    const data = await response.json();
    return data.results;
  }
  
  export async function fetchAvailableSorts(
    searchQuery: string,
    limit: number,
    selectedSort: string | null
  ) {
    const sortQueryParam = selectedSort ? `&sort=${selectedSort}` : '';
  
    const response = await fetch(
      `https://api.mercadolibre.com/sites/MLA/search?q=${searchQuery}&limit=${limit}${sortQueryParam}`
    );
  
    if (!response.ok) {
      throw new Error('Error al cargar available_sorts');
    }
  
    const data = await response.json();
    return data.available_sorts;
  }
  
  export async function fetchAvailablePriceFilter(searchQuery: string, limit: number) {
    const response = await fetch(
      `https://api.mercadolibre.com/sites/MLA/search?q=${searchQuery}&limit=${limit}`
    );
  
    if (!response.ok) {
      throw new Error('Error al cargar available_filters');
    }
  
    const data = await response.json();
    return data.available_filters.find((filter: any) => filter.id === 'price');
  }
  */