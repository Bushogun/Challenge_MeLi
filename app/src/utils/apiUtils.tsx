import { useDispatch } from 'react-redux';
import { setProducts, setLoading, setError } from '@/src/store/productSlice';

export async function fetchProductData(
  dispatch: any,
  searchQuery: string,
  limit: string,
  selectedSort: string,
  priceFilter: string,
) {
  try {
    dispatch(setLoading(true));

    let sortQueryParam = '';
    let priceQueryParam = '';
    if (selectedSort) {
      sortQueryParam = `&sort=${selectedSort}`;
    }
    if (priceFilter) {
      priceQueryParam = `&price=${priceFilter}`;
    }

    const response = await fetch(
      `https://api.mercadolibre.com/sites/MLA/search?q=${searchQuery}&limit=${limit}${sortQueryParam}${priceQueryParam}`
    );
    if (!response.ok) {
      throw new Error('Error al cargar productos');
    }

    const data = await response.json();
    dispatch(setLoading(false));

    return data;
  } catch (error) {
    dispatch(setLoading(false));
    throw error;
  }
}
