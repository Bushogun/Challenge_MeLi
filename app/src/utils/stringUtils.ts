export function capitalizeLocations(text: string): string {
    return text.replace(/\b[A-Z]+\b/g, (word) => {
        return word.charAt(0) + word.slice(1).toLowerCase();
    });
}

export function capitalizeTitle(text: string): string {
    return text
        .toLowerCase()
        .split(' ')
        .map((word) => word.charAt(0).toUpperCase() + word.slice(1))
        .join(' ');
}
