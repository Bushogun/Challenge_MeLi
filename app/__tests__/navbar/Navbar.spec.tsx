import { Navbar } from "@/src/components/navbar/Navbar";

import { useRouter } from "next/router";
import { renderWithProviders } from "@/src/utils/test-utils";

describe("Navbar component", () => {
  jest.mock("next/router", () => ({
    useRouter: jest.fn(),
  }));

  it("should render correctly ", () => {
    const { container } = renderWithProviders(<Navbar />);
    expect(container).toMatchSnapshot();
  });
});
