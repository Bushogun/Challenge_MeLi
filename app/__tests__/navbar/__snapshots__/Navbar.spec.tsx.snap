// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar component should render correctly  1`] = `
<div>
  <nav
    class="navbar"
  >
    <a
      href="/"
    >
      <img
        alt="Logo Mercado Libre"
        data-nimg="1"
        decoding="async"
        fetchpriority="high"
        height="45"
        src="/_next/image?url=%2Flogo_ml.png&w=128&q=75"
        srcset="/_next/image?url=%2Flogo_ml.png&w=64&q=75 1x, /_next/image?url=%2Flogo_ml.png&w=128&q=75 2x"
        style="color: transparent;"
        title="Logo Mercado Libre"
        width="60"
      />
    </a>
    <form
      aria-label="form-search"
      class="container"
    >
      <div
        class="input-group"
      >
        <input
          alt="search product"
          class="form-control"
          name="search"
          placeholder="Buscar productos"
          required=""
          type="text"
          value=""
        />
        <div
          class="input-group-append"
        >
          <button
            class="btn btn-search"
            type="submit"
          >
            <svg
              fill="none"
              height="1em"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <circle
                cx="11"
                cy="11"
                r="8"
              />
              <line
                x1="21"
                x2="16.65"
                y1="21"
                y2="16.65"
              />
            </svg>
          </button>
        </div>
      </div>
    </form>
  </nav>
</div>
`;
